cluster-provision/k8s/1.25-fips/provision.sh

```diff
@@ -0,0 +1,190 @@
+
+# Required for iscsi demo to work.
+yum -y install iscsi-initiator-utils
+
+# for rook ceph
+dnf -y install lvm2
+# Convince ceph our storage is fast (not a rotational disk)
+echo 'ACTION=="add|change", SUBSYSTEM=="block", KERNEL=="vd[a-z]", ATTR{queue/rotational}="0"' \
+	> /etc/udev/rules.d/60-force-ssd-rotational.rules
+
+# To prevent preflight issue related to tc not found
+dnf install -y iproute-tc
+# Install istioctl
+export PATH=$ISTIO_BIN_DIR:$PATH
+(
+  set -E
+  mkdir -p $ISTIO_BIN_DIR
+  curl https://storage.googleapis.com/kubevirtci-istioctl-mirror/istio-$ISTIO_VERSION/bin/istioctl -o $ISTIO_BIN_DIR/istioctl
+  chmod +x $ISTIO_BIN_DIR/istioctl
+)
+
+export CRIO_VERSION=1.24
+cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable.repo
+[devel_kubic_libcontainers_stable]
+name=Stable Releases of Upstream github.com/containers packages (CentOS_8_Stream)
+type=rpm-md
+baseurl=https://storage.googleapis.com/kubevirtci-crio-mirror/devel_kubic_libcontainers_stable/
+gpgcheck=0
+enabled=1
+EOF
+cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}.repo
+[devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}]
+name=devel:kubic:libcontainers:stable:cri-o:${CRIO_VERSION} (CentOS_8_Stream)
+type=rpm-md
+baseurl=https://storage.googleapis.com/kubevirtci-crio-mirror/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}
+gpgcheck=0
+enabled=1
+EOF
+if [[ "$release" == "centos8" ]]; then
+    dnf install -y cri-o containers-common-1-23.module_el8.7.0+1106+45480ee0.x86_64
+elif [[ "$release" == "centos9" ]]; then
+    dnf install -y cri-o
+fi
+
+echo "" >> /etc/containers/policy.json
+
+systemctl enable --now crio
+
+# install podman for functionality missing in crictl (tag, etc)
+dnf install -y podman
+dnf install -y libseccomp-devel
+
+# link docker to podman as we need docker in test repos to pre-pull images
+# don't break them by doing a symlink
+ln -s /usr/bin/podman /usr/bin/docker
+
+cat << EOF > /etc/containers/registries.conf
+[registries.search]
+registries = ['registry.access.redhat.com', 'registry.fedoraproject.org', 'quay.io', 'docker.io']
+
+[registries.insecure]
+registries = ['registry:5000']
+
+[registries.block]
+registries = []
+EOF
+
+#TODO: el8 repo
+# Add Kubernetes repository.
+cat <<EOF >/etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes
+baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=0
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOF
+
+# Install Kubernetes packages.
+dnf install --skip-broken --nobest --nogpgcheck --disableexcludes=kubernetes -y \
+    kubeadm-${version} \
+    kubelet-${version} \
+    kubectl-${version} \
+    kubernetes-cni
+
+kubeadm config images pull --kubernetes-version ${version}
+
+dnf install -y centos-release-nfv-openvswitch
+dnf install -y openvswitch2.16
+
+fips-mode-setup --enable
```

Comment:
Can't we just enable this inside the default provider with an ENV var?

Reply:
I wish, but this command requires a reboot, so I guess not?

Reply:
I don't remember that we reboot in the process of building the nodes before freezing them, so did you add that somewhere. Or is my memory failing me?

Reply:
@dhiller I think we could run this in the run stage and I don't see reboot as a problem.

Reply:
@xpivarc do you mean we don't need to provision a new provider, and can just re-use an existing one, like the cgroupsv2 lane does for example?
I'm not sure what you mean by "run stage", but if there's a way to avoid creating a new image we should definitely use it!